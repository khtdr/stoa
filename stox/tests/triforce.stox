#!/usr/bin/env stox

fun triforce() {
  print "triforce:
 ▲
▲ ▲";
  return;
}

triforce();

var i = 0;
var j = 0;

while (true)
  while (i < 7) {
    j = i;
    i = i + 1;
    while (j < 7) {
      j = j + 1;
      if (j == 1) continue;
      if (j == 2) break;
      print "i,j " + i + "," + j;
      if (j == 4) continue 2;
      print "-";
      if (j == 5) break;
      if (j == 6) break 3;
    }
    print "X";
  }

print "done!";
