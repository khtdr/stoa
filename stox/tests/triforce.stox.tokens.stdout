FUN[3,1]
IDENTIFIER[3,5]
LEFT_PAREN[3,13]
RIGHT_PAREN[3,14]
LEFT_CURL[3,16]
PRINT[4,3]
STRING[4,9]
SEMICOLON[6,5]
RETURN[7,3]
SEMICOLON[7,9]
RIGHT_CURL[8,1]
IDENTIFIER[10,1]
LEFT_PAREN[10,9]
RIGHT_PAREN[10,10]
SEMICOLON[10,11]
VAR[12,1]
IDENTIFIER[12,5]
EQUAL[12,7]
NUMBER[12,9]
SEMICOLON[12,10]
VAR[13,1]
IDENTIFIER[13,5]
EQUAL[13,7]
NUMBER[13,9]
SEMICOLON[13,10]
WHILE[15,1]
LEFT_PAREN[15,7]
TRUE[15,8]
RIGHT_PAREN[15,12]
WHILE[16,3]
LEFT_PAREN[16,9]
IDENTIFIER[16,10]
LESS[16,12]
NUMBER[16,14]
RIGHT_PAREN[16,15]
LEFT_CURL[16,17]
IDENTIFIER[17,5]
EQUAL[17,7]
IDENTIFIER[17,9]
SEMICOLON[17,10]
IDENTIFIER[18,5]
EQUAL[18,7]
IDENTIFIER[18,9]
PLUS[18,11]
NUMBER[18,13]
SEMICOLON[18,14]
WHILE[19,5]
LEFT_PAREN[19,11]
IDENTIFIER[19,12]
LESS[19,14]
NUMBER[19,16]
RIGHT_PAREN[19,17]
LEFT_CURL[19,19]
IDENTIFIER[20,7]
EQUAL[20,9]
IDENTIFIER[20,11]
PLUS[20,13]
NUMBER[20,15]
SEMICOLON[20,16]
IF[21,7]
LEFT_PAREN[21,10]
IDENTIFIER[21,11]
EQUAL_EQUAL[21,13]
NUMBER[21,16]
RIGHT_PAREN[21,17]
CONTINUE[21,19]
SEMICOLON[21,27]
IF[22,7]
LEFT_PAREN[22,10]
IDENTIFIER[22,11]
EQUAL_EQUAL[22,13]
NUMBER[22,16]
RIGHT_PAREN[22,17]
BREAK[22,19]
SEMICOLON[22,24]
PRINT[23,7]
STRING[23,13]
PLUS[23,20]
IDENTIFIER[23,22]
PLUS[23,24]
STRING[23,26]
PLUS[23,30]
IDENTIFIER[23,32]
SEMICOLON[23,33]
IF[24,7]
LEFT_PAREN[24,10]
IDENTIFIER[24,11]
EQUAL_EQUAL[24,13]
NUMBER[24,16]
RIGHT_PAREN[24,17]
CONTINUE[24,19]
NUMBER[24,28]
SEMICOLON[24,29]
PRINT[25,7]
STRING[25,13]
SEMICOLON[25,16]
IF[26,7]
LEFT_PAREN[26,10]
IDENTIFIER[26,11]
EQUAL_EQUAL[26,13]
NUMBER[26,16]
RIGHT_PAREN[26,17]
BREAK[26,19]
SEMICOLON[26,24]
IF[27,7]
LEFT_PAREN[27,10]
IDENTIFIER[27,11]
EQUAL_EQUAL[27,13]
NUMBER[27,16]
RIGHT_PAREN[27,17]
BREAK[27,19]
NUMBER[27,25]
SEMICOLON[27,26]
RIGHT_CURL[28,5]
PRINT[29,5]
STRING[29,11]
SEMICOLON[29,14]
RIGHT_CURL[30,3]
PRINT[32,1]
STRING[32,7]
SEMICOLON[32,14]
